tests.py

```diff
@@ -436,40 +643,403 @@ def __getitem__(self, key):
+            'id': str(uuid.uuid4()),
+            'name': 'user_name',
+            'disable_toopher_auth': False
+        }
+        self.user_id = self.user['id']
+        self.user_name = self.user['name']
+
     def test_incomplete_response_raises_exception(self):
         response = {'key': 'value'}
         def fn():
-            toopher.PairingStatus(response)
+            toopher.Pairing(response, self.api)
         self.assertRaises(toopher.ToopherApiError, fn)
 
     def test_nonzero_when_granted(self):
         response = ddict()
         response['enabled'] = True
-        allowed = toopher.PairingStatus(response)
+        allowed = toopher.Pairing(response, self.api)
         self.assertTrue(allowed)
 
         response['enabled'] = False
-        denied = toopher.PairingStatus(response)
+        denied = toopher.Pairing(response, self.api)
         self.assertFalse(denied)
 
+    def test_refresh_from_server(self):
+        response = {
+            'id': self.id,
+            'enabled': True,
+            'pending': True,
+            'user': self.user
+        }
+        pairing = toopher.Pairing(response, self.api)
+
+        self.api.advanced.raw.client = HttpClientMock({
+            'pairings/{0}'.format(pairing.id): (200,
+                json.dumps({
+                    'id': self.id,
+                    'enabled': False,
+                    'pending': False,
+                    'user': {
+                        'id': self.user_id,
+                        'name': 'user_name changed',
+                        'disable_toopher_auth': True
+                    }
+                })
+            )
+        })
+        pairing.refresh_from_server()
+        self.assertEqual(self.api.advanced.raw.client.last_called_method, 'GET')
+        self.assertEqual(pairing.id, self.id)
+        self.assertEqual(pairing.user.id, self.user_id)
+        self.assertEqual(pairing.user.name, 'user_name changed')
+        self.assertTrue(pairing.user.disable_toopher_auth)
+        self.assertFalse(pairing.enabled)
+        self.assertFalse(pairing.pending)
+
+    def test_get_qr_code_image(self):
+        response = {'id': self.id,
+                    'enabled': True,
+                    'pending': True,
+                    'user': self.user }
+        pairing = toopher.Pairing(response, self.api)
+
+        qr = qrcode.QRCode()
+        qr.add_data('https://api.toopher.test/v1/r/pairings/{0}'.format(self.id))
+        qr.make()
+        image = qr.make_image()
+        fp = StringIO()
+        image.save(fp, 'png')
+        qr_code_image = fp.getvalue()
+        fp.close
+
+        self.api.advanced.raw.client = HttpClientMock({
+            'qr/pairings/{0}'.format(pairing.id): (200,
+                    qr_code_image)
+        })
+        qr_image_data = pairing.get_qr_code_image()
+        self.assertEqual(self.api.advanced.raw.client.last_called_method, 'GET')
+        self.assertEqual(imghdr.what(None, qr_image_data), 'png')
+
```

Comment:
I took a stab at refactoring this test so that we're creating a QR code and using [StringIO](https://docs.python.org/2/library/stringio.html) to read it. This will eliminate the need to save qr_image.png to the repo since StringIO implements a file-like class and doesn't require an actual image file. There doesn't seem to be an easy way to [decode QR code images in python](http://stackoverflow.com/questions/27233351/how-to-decode-a-qr-code-image-in-preferably-pure-python), so for now I'm just testing the image type and last_called_method.